html/files/knowledge_panel_test.json

```diff
@@ -0,0 +1,91 @@
+                }
+            }
+        ]
+    },
+    "ecoscore": {
+        "type": "ecoscore",
+        "level": "info",
+        "topics": [
+            "environment"
+        ],
+        "title": "Eco-Score D",
+        "subtitle": "High environmental impact",
+        "grade": "d",
+        "icon_url": "https://static.openfoodfacts.org/images/attributes/ecoscore-d.svg",
+        "elements": [
+            {
+                "element_type": "text",
+                "element": {
+                    "text_type": "summary",
+                    "html": "The <a href=\"https://world.openfoodfacts.org/eco-score-the-environmental-impact-of-food-products\">Eco-Score</a> is an experimental score that summarizes the environmental impacts of food products."    
+                }
+            },
+            {
+                "element_type": "text",
+                "element": {
+                    "text_type": "note",
+                    "html": "The Eco-Score was initially developped for France and it is being extended to other European countries. The Eco-Score formula is subject to change as it is regularly improved to make it more precise and better suited to each country."    
+                }
+            },
+            {
+                "element_type": "panel",
+                "element": {
+                    "panel_id": "ecoscore_lca"
+                }  
+            },
+            {
+                "element_type": "text",
+                "element": {
+                    "text_type": "warning",
+                    "html": "<strong>Warning: some information necessary to calculate the Eco-Score with precision is not provided (see the details of the calculation below). You can edit the product to add the missing information.<br><br>If you are the manufacturer of this product, you can send us the information with our <a href=\"https://world.pro.openfoodfacts.org\">free platform for producers</a>."
+                }
+            }
+        ]
+    },
+    "ecoscore_lca" : {
+        "type": "ecoscore_lca",
+        "level": "info",
+        "title": "Lifecyle Analysis (LCA) for the product category",
+        "subtitle": "Score: 40/100",
+        "grade": "b",
+        "elements": [
+            {
+                "element_type": "text",
+                "element": {
+                    "text_type": "summary",
+                    "html": "<p>Agribalyse category: <a href=\"https://www.agribalyse.fr/app/aliments/31032\" title=\"Exact match with the product category\">Chocolate spread with hazelnuts</a></p><ul><li>PEF environmental score: 0.74 (the lower the score, the lower the impact)</li><li>- including impact on climate change: 9.87 kg CO2 eq/kg of product</li></ul>"
+                }
+            },
+            {
+                "element_type": "table",
+                "element": {
+                    "table_type": "percents",
```

Comment:
can we call this something more descriptive, like ecoscore_impact_breakdown ?

Reply:
We can add a "table_id" field to tables so that the client can identify a specific table.

Reply:
got it, so my understanding is that table_type: "percents" should have 2 headers and each row should contain a title and value.

Can we document this, clients would be happy to follow the documentation to understand the structure.

Reply:
mmm actually, it would be better if we can just create generic tables, right now I have to create a specific object for table_2_columns and potentially another object for other tables.

It would be better to create a generic_table type, with N headers and N values (value1, value2 ... valueN). If we want to distinguish some special tables (like percent) that can also be rendered as pie chart/graphs etc, we can use "table_type" for it, like "percent" here. This should accomplish all use cases you have in mind (AFAICT) and is easier to implement on the client side.

if you want, we can chat more on this, I can show you why the code is less complicated with generic tables

Reply:
so my suggestion is to turn the rows into an array, something like this:

'table_type': 'percents',
'headers': ['Steps', 'Impact'],
'rows': [
  ['Agriculture', '82.7'],
  ['Processing', '11.5'],
  ['Packaging', '2.8'],
  ['Transportation', '2.4'],
  ['Distribution', '0.6'],
  ['Consumption', '0.0']
]

Since the table_type is percents, the client should be able to decipher that it can also render this table as a bar/spider-graph/pie-chart etc.

Reply:
We can use an array for the column values indeed, but I still think we need properties for rows.
Here is what I propose:

```
                    "rows": [
                        {"id": "agriculture", "icon_url": "https://static.openfoodfacts.org/images/icons/dist/agriculture.svg", "values": ["Agriculture", "82.7%"], "percent": 82.7},
                        {"id": "processing", "values": ["Processing", "11.5%"], "percent": 11.5},	
                        {"id": "packaging", "values": ["Packaging", "2.8%"], "percent": 2.8},
                        {"id": "transportation", "values": ["Transportation", "2.4%"], "percent": 2.4},
                        {"id": "distribution", "values": ["Distribution", "0.6%"], "percent": 0.6},
                        {"id": "consumption", "values": ["Consumption", "0.0%"], "percent": 0}
                    ]

```
each row has optional properties, and at least a "values" array.

The properties can be used to pass icons etc.

We can also use them to pass the percent info as a number. We need both a string and a number, because the string value we display will be translated (e.g. in French we will have "75,2%" with a comma, and in English it will be "72.5%" with a dot.

Reply:
Sounds good, this looks better, thanks Stephane.